R/output_combined_table.R

```diff
@@ -0,0 +1,137 @@
+      dplyr::mutate(location_properties = paste0(location_property, "=", value)) %>%
+      dplyr::select(dplyr::all_of(c("dataset_id", "location_id", "location_name", "location_properties"))) %>%
+      dplyr::group_by(dataset_id, location_id, location_name) %>%
+      dplyr::mutate(location_properties = paste0(location_properties, collapse = "; ")) %>%
+      dplyr::ungroup() %>%
+      dplyr::distinct()
+
+  contributors <-
+    database$contributors %>%
+      dplyr::mutate(
+        affiliation = stringr::str_replace_all(affiliation, ":", "-"),
+        affiliation = stringr::str_replace_all(affiliation, ";", ","),
+        affiliation = stringr::str_replace_all(affiliation, "<", "("),
+        affiliation = stringr::str_replace_all(affiliation, ">", ")"),
+        additional_role = stringr::str_replace_all(additional_role, "<", "("),
+        additional_role = stringr::str_replace_all(additional_role, ">", ")"),
+        data_collectors = paste0(given_name, " ", last_name),
+        data_collectors = ifelse(
+          !is.na(ORCID),
+          paste0(data_collectors, " <ORCID:", ORCID),
+          data_collectors),
+        data_collectors = ifelse(
+          is.na(ORCID),
+          paste0(data_collectors, " <affiliation:", affiliation),
+          paste0(data_collectors, ";affiliation:", affiliation)),
+        data_collectors = ifelse(
+          !is.na(additional_role),
+          paste0(data_collectors, ";additional_role:", additional_role, ">"),
+          paste0(data_collectors, ">"))
+      ) %>%
+      dplyr::select(-dplyr::all_of(c("last_name", "given_name", "ORCID", "affiliation", "additional_role"))) %>%
+      dplyr::group_by(dataset_id) %>%
+      dplyr::mutate(data_collectors = paste0(data_collectors, collapse = "; ")) %>%
+      dplyr::ungroup() %>%
+      dplyr::distinct()
+
+  contexts_tmp <-
+    database$contexts %>%
+      dplyr::mutate(
+        context_property = stringr::str_replace_all(context_property, "=", "-"),
+        value = stringr::str_replace_all(value, "=", "-"),
+        description = stringr::str_replace_all(description, "=", "-"),
+        context_property = stringr::str_replace_all(context_property, ";", ","),
+        value = stringr::str_replace_all(value, ";", ","),
+        description = stringr::str_replace_all(description, "=", "-"),
+        value = ifelse(
+          is.na(description),
+          paste0(context_property, ":", value),
+          paste0(context_property, ":", value, " <", description, ">"))
+      ) %>%
+    dplyr::select(-dplyr::all_of(c("description", "context_property", "category"))) %>%
+    tidyr::separate_longer_delim(link_vals, ", ") %>% distinct()
+
+  reformat_contexts <- function(contexts_table, context_id) {
+    context_category <- gsub("_id", "_properties", context_id, fixed = TRUE)
+    out <- contexts_table %>%
+      dplyr::filter(link_id == context_id) %>%
+      dplyr::select(-link_id) %>%
+      dplyr::distinct(dataset_id, link_vals, .keep_all = TRUE)
```

Comment:
Is this line necessary? Shouldn't it be distinct anyway, otherwise there's something wrong with the contexts table that we want to be picked up by `dataset_test`?